AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,104 @@
 //
 
 import UIKit
+import AVKit
+import AVFoundation
+import MobileCoreServices
 
-class DoubleVideoViewController: UIViewController, CellTitled {
+class DoubleVideoViewController: UIViewController, CellTitled, UIImagePickerControllerDelegate, UINavigationControllerDelegate, AVPlayerViewControllerDelegate {
     var titleForCell: String = "Double Video"
 
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
+    var movieURL: URL?
+    var playerTop = AVPlayer()
+    var playerLayerTop: AVPlayerLayer!
+    var playerBottom = AVPlayer()
+    var playerLayerBottom: AVPlayerLayer!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        // don't f with this 👇 comment
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
     }
 
     override func viewDidLayoutSubviews() {
-        // update your layers' frames here
+        guard let sublayersTop = self.videoContainerTop.layer.sublayers else { return }
+        
+        for layer in sublayersTop {
+            layer.frame = self.videoContainerTop.bounds
+        }
+        
+        guard let sublayersBottom = self.videoContainerBottom.layer.sublayers else { return }
+        
+        for layer in sublayersBottom {
+            layer.frame = self.videoContainerBottom.bounds
+        }
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.sourceType = .photoLibrary
+        imagePickerController.mediaTypes = [String(kUTTypeMovie)]
+        imagePickerController.delegate = self
+        self.present(imagePickerController, animated: true, completion: nil)
+    }
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        switch info[UIImagePickerControllerMediaType] as! String {
+        case String(kUTTypeImage):
+            if let image = info[UIImagePickerControllerOriginalImage] as? UIImage {
+                print("BAKANAAAA!!!: \(image)")
+            }
+        case String(kUTTypeMovie):
+            if let url = info[UIImagePickerControllerMediaURL] as? URL {
+                self.movieURL = url
+            }
+        default:
+            print("muzukashi na~")
+        }
+        
+        // dismissing imagePickerController
+        dismiss(animated: true) {
+//            if let url = self.movieURL {
+//                let player = AVPlayer(url: url)
+//                let playerController = AVPlayerViewController()
+//                playerController.player = player
+//                self.present(playerController, animated: true, completion: {
+//                    self.movieURL = nil
+//                })
+//            }
+            if self.playerTop.rate == 0.0 {
```

Comment:
The one bonus you didn't do was to put these in a "data structure" i.e. an array. Two players are manageable but an array would lead toward having many open.